pkg/crc/machine/hyperv/driver_windows.go

```diff
@@ -23,5 +29,32 @@ func CreateHost(machineConfig config.MachineConfig) *hyperv.Driver {
 		hypervDriver.VirtualSwitch = switchName
 	}
 
+	for _, dir := range machineConfig.SharedDirs {
+		sharedDir := drivers.SharedDir{
+			Source:   dir,
+			Target:   convertToUnixPath(dir),
+			Tag:      smbShareUNCPathFromTag("crc-dir0"), // smb share 'crc-dir0' is created in the msi
+			Type:     "cifs",
+			Username: machineConfig.SharedDirUsername,
```

Comment:
so we associate the username at this point, but not the password? Why do we need to do this in the previous part in an update function?

Reply:
because we don't want to store the password in the machine config file, so that's handled by the update function

Reply:
If you are concerned about the password, this is an association as a credential (username + password). It would make more sense to place the username in the update function too.

Reply:
Its not as much about leaking information but storing a secret on disk unencrypted which we want to prevent, on the user's machine any non-privilege process can just run `whoami` and get that username, so it should be fine to store in the machine config?

btw this was suggested in https://github.com/code-ready/crc/pull/3308#discussion_r953798210

Reply:
> any non-privilege process can just run whoami and get that username, 

People share the virtual machine folder as we do not offer an export function.

Reply:
ahh, but that'd be a very special unsupported use case, assume most of our users don't do this, also we don't test it

Reply:
It turns out to be different from what I suggested earlier, but maybe this should all be coming from the keyring then?

Reply:
only the password needs to be securely stored, username we can derive during start, we just skip writing it to machine config json same as the password